garrysmod/lua/entities/sent_ball.lua

```diff
@@ -141,9 +141,9 @@ end
 
 function ENT:Use( activator, caller )
 
-	self:Remove()
+	if ( activator:IsPlayer() ) and ( ( game.SinglePlayer() and CLIENT ) or SERVER ) then

```

Comment:
Just change this check to SERVER, entities should never be removed on the client.